src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -69,164 +174,326 @@ static unsafe BoxedValueIReferenceImpl()
-        public static implicit operator Nullable<T>(ObjectReference<Vftbl> obj) => (obj != null) ? new Nullable<T>(obj) : null;
-        protected readonly ObjectReference<Vftbl> _obj;
-        public IntPtr ThisPtr => _obj.ThisPtr;
-        public ObjectReference<I> AsInterface<I>() => _obj.As<I>();
-        public A As<A>() => _obj.AsType<A>();
-        public Nullable(IObjectReference obj) : this(obj.As<Vftbl>(Vftbl.PIID)) { }
-        public Nullable(ObjectReference<Vftbl> obj)
-        {
-            _obj = obj;
+            ComWrappersSupport.CreateCCWForObjectForMarshaling(value, PIID);
+
+        public static IntPtr GetAbi(IObjectReference m) => m?.ThisPtr ?? IntPtr.Zero;
+
+        public static object FromAbi(IntPtr ptr)
+        {
+            if (ptr == IntPtr.Zero)
+            {
+                return null;
+            }
+            var vftblT = new Vftbl(ptr);
+            var wrapper = new Nullable<T>(ObjectReference<Vftbl>.FromAbi(ptr, vftblT));
+            return wrapper.Value;
+        }
+
+        public static unsafe void CopyManaged(object o, IntPtr dest)
+        {
+            *(IntPtr*)dest.ToPointer() = CreateMarshaler2(o).Detach();
+        }
+
+        public static IntPtr FromManaged(object value)
+        {
+            if (value is null)
+            {
+                return IntPtr.Zero;
+            }
+            return CreateMarshaler2(value).Detach();
+        }
+
+        public static void DisposeMarshaler(IObjectReference m) { m?.Dispose(); }
+        public static void DisposeAbi(IntPtr abi) { MarshalInspectable<object>.DisposeAbi(abi); }
+
+        public static string GetGuidSignature() => CreateGuidSignature();
+
+        private static string CreateGuidSignature()
+        {
+            if (typeof(T) == typeof(int)) return Nullable_int.GetGuidSignature();
+            if (typeof(T) == typeof(byte)) return Nullable_byte.GetGuidSignature();
+            if (typeof(T) == typeof(bool)) return Nullable_bool.GetGuidSignature();
+            if (typeof(T) == typeof(sbyte)) return Nullable_sbyte.GetGuidSignature();
+            if (typeof(T) == typeof(short)) return Nullable_short.GetGuidSignature();
+            if (typeof(T) == typeof(ushort)) return Nullable_ushort.GetGuidSignature();
+            if (typeof(T) == typeof(char)) return Nullable_char.GetGuidSignature();
+            if (typeof(T) == typeof(uint)) return Nullable_uint.GetGuidSignature();
+            if (typeof(T) == typeof(long)) return Nullable_long.GetGuidSignature();
+            if (typeof(T) == typeof(ulong)) return Nullable_ulong.GetGuidSignature();
+            if (typeof(T) == typeof(float)) return Nullable_float.GetGuidSignature();
+            if (typeof(T) == typeof(double)) return Nullable_double.GetGuidSignature();
+            if (typeof(T) == typeof(Guid)) return Nullable_guid.GetGuidSignature();
+            if (typeof(T) == typeof(global::System.TimeSpan)) return Nullable_TimeSpan.GetGuidSignature();

```

Comment:
There is also technically `Nullable_Type`.

Reply:
Added to the signature switch for good measure. Didn't wire up the rest since we mentioned we're not even entirely sure that this code is actually ever being executed (though it is being compiled, that part we're sure of). We can revisit this in a follow up PR and add that complexity if needed I think 🙂